var/spack/repos/builtin/packages/libcumlprims/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from spack import *
+
+
+class Libcumlprims(Package):
+    """libcuMLPrims library"""
+
+    homepage = "https://rapids.ai"
+    url      = "https://anaconda.org/nvidia/libcumlprims/0.15.0/download/linux-64/libcumlprims-0.15.0-cuda11.0_gdbd0d39_0.tar.bz2"
+
+    version('0.15.0-cuda11.0_gdbd0d39_0', sha256='0edc55767f06f533fbff7a0fecaf6e6d4f82eec39604b3874a07b5609f79ece8')
+    version('0.15.0-cuda10.2_gdbd0d39_0', sha256='b7a8740de0d15380829f42fcb078567e73ab7d29b14be073376153bf2d8ec945')
+    version('0.15.0-cuda10.1_gdbd0d39_0', sha256='f055f904b5ef67995869b0bc648d9fe30839b08e77cb335573bf9f1c816d4d9b')
+
+    depends_on('cuda@11.0.0:11.0.999', when='@0.15.0-cuda11.0_gdbd0d39_0')
+    depends_on('cuda@10.2.0:10.2.999', when='@0.15.0-cuda10.2_gdbd0d39_0')
+    depends_on('cuda@10.1.0:10.1.999', when='@0.15.0-cuda10.1_gdbd0d39_0')
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.prepend_path('SPACK_INCLUDE_DIRS', self.prefix.include.cumlprims)
```

Comment:
`SPACK_INCLUDE_DIRS` is generally for Spack's internal use: could you implement a `headers` property which includes this directory?

In `freetype` this looks like:

```
    @property
    def headers(self):
        headers = find_headers('*', self.prefix.include, recursive=True)
        headers.directories = [self.prefix.include.freetype2]
        return headers
```

I don't think the part that sets `headers.directories` is needed so in your case you can just do:

```
    @property
    def headers(self):
        headers = find_headers('*', self.prefix.include, recursive=True)
        return headers
```

Reply:
I replaced lines 24 and 25 with:

```python
    @property
    def headers(self):
        headers = find_headers('*', self.prefix.include, recursive=True)
        headers.directories = [self.prefix.include.cumlprims]
        return headers
```

and dependent packages aren't able to find the headers they're expecting to find. Am I missing something?

Reply:
Sanity check: do those dependent packages refer to this as a `link` dependency? If you don't specify that in the `depends_on`, it will include `link` by default. If you have specified `link` (or haven't set `depends_on(..., type=)` at all) then what you did should work, and I need to investigate why it isn't working.

Reply:
They don't specify which type

Reply:
Fixed